pkg/runtime/runtime.go

```diff
@@ -3150,14 +3297,46 @@ func ShutdownSignal() chan os.Signal {
 	return stop
 }
 
+func (a *DaprRuntime) createHTTPEndpointsChannels() (map[string]channel.HTTPEndpointAppChannel, error) {
+	// Create dedicated app channels for known app endpoints
+	endpoints := a.compStore.ListHTTPEndpoints()
+
+	if len(endpoints) > 0 {
+		channels := make(map[string]channel.HTTPEndpointAppChannel, len(endpoints))
+
+		for _, e := range endpoints {
+			pipeline, err := a.buildAppHTTPPipeline()
+			if err != nil {
+				return nil, fmt.Errorf("failed to build app HTTP pipeline: %w", err)
+			}
+
+			conf, err := a.getHTTPEndpointAppChannel(pipeline, e)
+			if err != nil {
+				return nil, err
+			}
+
+			ch, err := httpChannel.CreateHTTPChannel(conf)
+			if err != nil {
+				return nil, err
+			}
+
+			channels[e.ObjectMeta.Name] = ch
```

Comment:
Thoughts on perhaps initializing this map lazily to be easier on the memory? (although then we'd have to worry about concurrent access among goroutines…)

Reply:
Yes, lazy initializing will provide minuscule benefits here.